In [3]:
import pandas as pd
import numpy as np  

print('Pandas Version Nro:\n', {pd.__version__})
print('NumPy Version Nro:\n', {np.__version__})

Pandas Version Nro:
 {'2.3.1'}
NumPy Version Nro:
 {'2.3.2'}


In [6]:
df_videogames_repaso = pd.read_csv('vgsales.csv')

df_videogames_repaso.head()

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
0,1,Wii Sports,Wii,2006.0,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,2,Super Mario Bros.,NES,1985.0,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,3,Mario Kart Wii,Wii,2008.0,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,4,Wii Sports Resort,Wii,2009.0,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,5,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37


In [7]:
df_videogames_repaso.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


Se observan en una primera impresion valores nulos en las columnas de 'Year' y 'Publisher', lo que puede afectar el analisis de los datos.

In [8]:
df_videogames_repaso.describe()

,Rank,Year,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
count,16598.000000,16327.000000,16598.000000,16598.000000,16598.000000,16598.000000,16598.000000
mean,8300.605254,2006.406443,0.264667,0.146652,0.077782,0.048063,0.537441
std,4791.853933,5.828981,0.816683,0.505351,0.309291,0.188588,1.555028
min,1.000000,1980.000000,0.000000,0.000000,0.000000,0.000000,0.010000
25%,4151.250000,2003.000000,0.000000,0.000000,0.000000,0.000000,0.060000
50%,8300.500000,2007.000000,0.080000,0.020000,0.000000,0.010000,0.170000
75%,12449.750000,2010.000000,0.240000,0.110000,0.040000,0.040000,0.470000
max,16600.000000,2020.000000,41.490000,29.020000,10.220000,10.570000,82.740000


Se obverva que la desviacion estandar de las ventas globales es muy alta, lo que indica que hay una gran variabilidad en las ventas de los videojuegos. Esto sugiere que algunos videojuegos se venden mucho mejor que otros, lo que puede deberse a factores como la popularidad del juego, la calidad del marketing, o la competencia en el mercado, tambien puede deberse a numeroras ventas por lanzamiento que lo que tambien podria afectar las ventas globales.

In [9]:
df_videogames_repaso.isna().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

Se confirman 271 datos nulos para la columna 'Year' y 58 para la columna 'Publisher', estos seran reemplazos con el el año promedio para la columna 'Year' y 'Unknown' ya que a pesar de representar un valor muy porcentaje muy bajo en la base de datos, se asume que dicha informacion posee influencia en las ventas globales.

In [19]:
mediana_year = df_videogames_repaso['Year'].median()

print('Calculo de la mediana de la columna Year:\n',{mediana_year})

Calculo de la mediana de la columna Year:
 {np.float64(2007.0)}


In [20]:
df_videogames_repaso['Year'] = df_videogames_repaso['Year'].fillna(mediana_year)

In [21]:
df_videogames_repaso['Publisher'] = df_videogames_repaso['Publisher']. fillna('Unknown')

In [22]:
df_videogames_repaso.isna().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

In [23]:
df_videogames_repaso.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16598 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16598 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [25]:
df_videogames_repaso['Year'] = df_videogames_repaso['Year'].astype(int)

df_videogames_repaso.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16598 non-null  int64  
 4   Genre         16598 non-null  object 
 5   Publisher     16598 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(5), int64(2), object(4)
memory usage: 1.4+ MB


In [26]:
df_videogames_repaso['Sales_category'] = df_videogames_repaso['Global_Sales'].apply(lambda sales: 'Existo de ventas' if sales > 1
                                                                                    else 'Ventas estandar')
df_videogames_repaso.head()

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales,Sales_category
0,1,Wii Sports,Wii,2006,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74,Existo de ventas
1,2,Super Mario Bros.,NES,1985,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24,Existo de ventas
2,3,Mario Kart Wii,Wii,2008,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82,Existo de ventas
3,4,Wii Sports Resort,Wii,2009,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00,Existo de ventas
4,5,Pokemon Red/Pokemon Blue,GB,1996,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37,Existo de ventas


In [34]:
print('Cantidad de filas duplicadas en el DataFrame:\n', {df_videogames_repaso.duplicated().sum()})

Cantidad de filas duplicadas en el DataFrame:
 {np.int64(0)}


In [35]:
print('Cantidad de filas duplicadas en las columnas Name y Platform:\n', {df_videogames_repaso.duplicated(subset=['Name','Platform']).sum()})

Cantidad de filas duplicadas en las columnas Name y Platform:
 {np.int64(5)}


Se confirman 5 duplicados en la base de datos, estos seran eliminados para evitar sesgos de la informacion.

In [36]:
df_videogames_repaso = df_videogames_repaso.drop_duplicates(subset=['Name','Platform'])

df_videogames_repaso.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16593 entries, 0 to 16597
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Rank            16593 non-null  int64  
 1   Name            16593 non-null  object 
 2   Platform        16593 non-null  object 
 3   Year            16593 non-null  int64  
 4   Genre           16593 non-null  object 
 5   Publisher       16593 non-null  object 
 6   NA_Sales        16593 non-null  float64
 7   EU_Sales        16593 non-null  float64
 8   JP_Sales        16593 non-null  float64
 9   Other_Sales     16593 non-null  float64
 10  Global_Sales    16593 non-null  float64
 11  Sales_category  16593 non-null  object 
dtypes: float64(5), int64(2), object(5)
memory usage: 1.6+ MB


In [38]:
hits_de_nintendo = df_videogames_repaso 

hits_de_nintendo = hits_de_nintendo[(hits_de_nintendo['Publisher'] == 'Nintendo') & (hits_de_nintendo['Global_Sales'] > 1)]

hits_de_nintendo = hits_de_nintendo.sort_values(by='Global_Sales', ascending=False)

hits_de_nintendo.head(10)

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales,Sales_category
0,1,Wii Sports,Wii,2006,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74,Existo de ventas
1,2,Super Mario Bros.,NES,1985,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24,Existo de ventas
2,3,Mario Kart Wii,Wii,2008,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82,Existo de ventas
3,4,Wii Sports Resort,Wii,2009,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00,Existo de ventas
4,5,Pokemon Red/Pokemon Blue,GB,1996,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37,Existo de ventas
5,6,Tetris,GB,1989,Puzzle,Nintendo,23.20,2.26,4.22,0.58,30.26,Existo de ventas
6,7,New Super Mario Bros.,DS,2006,Platform,Nintendo,11.38,9.23,6.50,2.90,30.01,Existo de ventas
7,8,Wii Play,Wii,2006,Misc,Nintendo,14.03,9.20,2.93,2.85,29.02,Existo de ventas
8,9,New Super Mario Bros. Wii,Wii,2009,Platform,Nintendo,14.59,7.06,4.70,2.26,28.62,Existo de ventas
9,10,Duck Hunt,NES,1984,Shooter,Nintendo,26.93,0.63,0.28,0.47,28.31,Existo de ventas


In [39]:
hits_de_nintendo.to_csv('hits_de_nintendo.csv', index=False)